src/lp/fixed_variable.jl

```diff
@@ -1,68 +1,177 @@
-            ps.lrow[i] -= aij * lb
-            ps.urow[i] -= aij * lb
-
-            ps.nzrow[i] -= 1
-
-            # Check row
-            if ps.nzrow[i] == 0
-                remove_empty_row!(ps, i)
-            elseif ps.nzrow == 1
-                push!(ps.row_singletons, i)
-            end
-        end  # row update
+    abs(ub - lb) <= ϵ || return nothing
+
+    @debug "Fixing variable $j to $lb"
+
+    # Infeasibility check for integer variables
+    vt = ps.pb0.var_types[j]
+    if is_integer(vt)
+        # Check if fixing to fractional value
+        f = min(lb - floor(lb), ceil(lb) - lb)
+        if f >= ϵ_int || (vt == BINARY
+            && ((lb <= -ϵ_int) || (lb >= 1 + ϵ_int))
+        )
+            # MIP problem is infeasible
+            ps.status = PRIMAL_INFEASIBLE
+            return nothing
+        end
     end
 
+    col = ps.pb0.acols[j]
+    cj = ps.obj[j]
+
+    # Remove column
+    ps.colflag[j] = false
+    ps.ncol -= 1
+    ps.updated = true
+
+    # TODO: make this more efficient
+    push!(ps.ops, FixedVariable(j, lb, cj, Col(
+        [i for i in col.nzind if ps.rowflag[i]],
+        [aij for (i, aij) in zip(col.nzind, col.nzval) if ps.rowflag[i]]
+    )))
+
+    # Update objective constant
+    ps.obj0 += cj * lb
+
+    # Update rows
+    # TODO: update list of row singletons and empty rows
+    for (i, aij) in zip(col.nzind, col.nzval)
+        ps.rowflag[i] || continue  # This row is no longer in the problem
+        iszero(aij) && continue  # Skip if coefficient is zero
+
+        # Update row bounds
+        ps.lrow[i] -= aij * lb
+        ps.urow[i] -= aij * lb
+
+        ps.nzrow[i] -= 1
+
+        # Check row
+        if ps.nzrow[i] == 0
+            remove_empty_row!(ps, i)
+        elseif ps.nzrow == 1
+            push!(ps.row_singletons, i)
+        end
+    end  # row update
+
     # Done
     return nothing
 end
 
-function postsolve!(sol::Solution{T}, op::FixedVariable{T}) where{T}
-    sol.x[op.j] = op.x
-    s = sol.is_dual_ray ? zero(T) : op.c
-    for (i, aij) in zip(op.col.nzind, op.col.nzval)
+function postsolve!(sol::Solution{T}, r::FixedVariable{T}) where{T}
+    sol.x[r.j] = r.x
+    s = sol.is_dual_ray ? zero(T) : r.c
+    for (i, aij) in zip(r.col.nzind, r.col.nzval)
         s -= aij * (sol.y_lower[i] - sol.y_upper[i])
     end
-    sol.s_lower[op.j] = pos_part(s)
-    sol.s_upper[op.j] = neg_part(s)
+    sol.s_lower[r.j] = pos_part(s)
+    sol.s_upper[r.j] = neg_part(s)
     return nothing
-end
\ No newline at end of file
+end
+
+"""
+    RemoveFixedVariables <: AbstractRule
+
+Remove all fixed variables in the problem.
+
+See [`FixSingleVariable`](@ref).
+"""
+struct RemoveFixedVariables <: AbstractRule end
+
+function apply!(
+    ps::PresolveData{T},
+    ::RemoveFixedVariables,
+    config::PresolveOptions{T}
+) where {T}
+
+    for (j, flag) in enumerate(ps.colflag)
+        apply!(ps, FixSingleVariable(j), config)
+        ps.status == NOT_INFERRED || return nothing
```

Comment:
Can use ``@_return_if_inferred`` here as well

Reply:
I didn't use it here because, currently, `@_return_if_inferred` will return `ps.status`, not `nothing`.

That's a small style discrepancy between `presolve!`, which returns the presolve status, and `apply!` which returns `nothing`.
If it needs discussing, I'd prefer to discuss it in another PR.